src/CHIPBindings.cc

```diff
@@ -705,9 +776,14 @@ hipError_t hipGraphMemcpyNodeGetParams(hipGraphNode_t node,
                                        hipMemcpy3DParms *pNodeParams) {
   CHIP_TRY
   CHIPInitialize();
-  hipMemcpy3DParms Params =
-      static_cast<CHIPGraphNodeMemcpy *>(node)->getParams();
-  pNodeParams = &Params;
+  if (!node || !pNodeParams)
+    RETURN(hipErrorInvalidValue);
+  auto CastNode = static_cast<CHIPGraphNodeMemcpy *>(node);
+  if (CastNode->getType() != hipGraphNodeTypeMemcpy)
+    CHIPERR_LOG_AND_THROW("Node is not Memcpy", hipErrorInvalidValue);
```

Comment:
if the cast failed, woudln't CastNode be null?

Reply:
As Henry mentioned above, `static_cast` either fails at compile time, or doesn't fail at all. Null from failed cast is only returned by `dynamic_cast`, but using that requires compiling all code with RTTI enabled. There is a graph test that tries to call some API with node object of wrong typed, and `static_cast` didn't return null. For this reason i changed these checks to call `getType()` and compare.